moviepy/audio/AudioClip.py

```diff
@@ -67,15 +67,14 @@ def iter_chunks(self, chunksize=None, chunk_duration=None, fps=None,
         
         totalsize = int(fps*self.duration)
 
-        if (totalsize % chunksize == 0):
-            nchunks = totalsize // chunksize
-        else:
-            nchunks = totalsize // chunksize + 1
+        nchunks = totalsize // chunksize + 1
 
-        pospos = list(range(0, totalsize,  chunksize))+[totalsize]
+        pospos = np.linspace(0, totalsize, nchunks + 1, endpoint=True, dtype=int)
```

Comment:
Yes. Since we're generating `endpoint=True`, we need to pass in one more than the number of chunks. I.e. if `nchunks` is 1,  we tell `linespace` to return two points (a start and an end).